plugins/org.jkiss.dbeaver.ui.editors.base/src/org/jkiss/dbeaver/ui/editors/MultiPageAbstractEditor.java

```diff
@@ -226,4 +221,44 @@ public void recreatePages() {
         }
         createPages();
     }
+
+    @Override
+    public void setFocus() {
+        final DBPPreferenceStore store = DBWorkbench.getPlatform().getPreferenceStore();
+        String storedScreenReader = store.getString(DatabaseEditorPreferences.PREF_SCREEN_READER_ACCESSIBILITY);
+        ScreenReader screenReader = ScreenReader.getScreenReader(storedScreenReader);
+        switch (screenReader) {
+            case JAWS:
+                if (activePageIndex != -1) {
+                    CTabItem tabItem = tabsList.get(activePageIndex);
+                    if (tabItem != null && !tabItem.isDisposed()) {
+                        if (tabItem.getControl() != null && !tabItem.getControl().isDisposed()) {
+                            tabItem.getControl().setFocus();
+                            tabItem.getParent().forceFocus();
+                        }
+                    }
+                }
+                break;
+            case NARRATOR:
+            case NVDA:
+            case OTHER:
+                if (activePageIndex != -1) {
+                    CTabItem tabItem = tabsList.get(activePageIndex);
+                    if (tabItem != null && !tabItem.isDisposed()) {
+                        if (tabItem != null && !tabItem.isDisposed()) {
+                            Control control = tabItem.getControl();
+                            if (control != null && !control.isDisposed()) {
+                                control.forceFocus();
+                            }
+                        }
+                    }
+                }
+                break;
+            case DEFAULT:
+                super.setFocus();
+                break;
+            default:
+                super.setFocus();
```

Comment:
I would highly encourage you to use new switch statements that require the programmer to introduce branches for every single variant of an enum. This way, no `default` branch is needed, and the compiler will complain about new variants if one or more will be added later.

Reply:
Example:

```java
switch (screenReader) {
    case JAWS -> { ... }
    case NARRATOR, NVDA, OTHER -> { ... }
    case DEFAULT -> { ... } // still not sure what's the "default" screen reader
}
```

Reply:
This case of extra duplicated code.

Reply:
What do you mean?

Reply:
missunderstood "...for every single variant of an enum" that looks your require to define each case as a signle block.